assets/src/domain/eventEditor/services/apollo/mutations/useMutationOptions.ts

```diff
@@ -16,62 +16,65 @@ import { useMutationHandler, mutations } from './';
+				return mutations[mutation];
+			};
 
-		/**
-		 * @param {string} mutationType Type of mutation
-		 * @param {object} input     Mutation input
-		 */
-		const getMutationOptions = (): OperationVariables => {
-			const mutationHandler = getMutationHandler(typeName);
 			/**
-			 * options = {
-			 *     variables,
-			 *     optimisticResponse,
-			 * 	   onUpdate,
-			 *     onCompleted,
-			 * 	   onError,
-			 * }
+			 * @param {string} mutationType Type of mutation
+			 * @param {object} input     Mutation input
 			 */
-			const { onUpdate, ...mutationOptions } = mutationHandler(mutationType, input);
+			const getMutationOptions = (): OperationVariables => {
+				const mutationHandler = getMutationHandler(typeName);
+				/**
+				 * options = {
+				 *     variables,
+				 *     optimisticResponse,
+				 * 	   onUpdate,
+				 *     onCompleted,
+				 * 	   onError,
+				 * }
+				 */
+				const { onUpdate, ...mutationOptions } = mutationHandler(mutationType, input);
 
-			let update: MutationUpdaterFn;
+				let update: MutationUpdaterFn;
 
-			if (typeof onUpdate === 'function') {
-				update = getUpdateCallback(onUpdate);
-			}
+				if (typeof onUpdate === 'function') {
+					update = getUpdateCallback(onUpdate);
+				}
 
-			return { ...mutationOptions, update };
-		};
+				return { ...mutationOptions, update };
+			};
 
-		const getUpdateCallback = (onUpdate: OnUpdateFn): MutationUpdaterFn => {
-			/**
-			 * Since every mutation update callback is interested
-			 * in the updated entity data in response, we will
-			 * pass just that entity to onUpdate.
-			 */
-			return (proxy: DataProxy, result: FetchResult): void => {
-				// e.g. "createDatetime", "updateTicket"
-				const mutationName = `${mutationType.toLowerCase()}Espresso${typeName}`;
-				// Example result: { data: { deletePrice: { price : {...} } } }
-				const path = ['data', mutationName, `espresso${typeName}`];
-				const entity = pathOr<any>({}, path, result);
+			const getUpdateCallback = (onUpdate: OnUpdateFn): MutationUpdaterFn => {
+				/**
+				 * Since every mutation update callback is interested
+				 * in the updated entity data in response, we will
+				 * pass just that entity to onUpdate.
+				 */
+				return (proxy: DataProxy, result: FetchResult): void => {
+					// e.g. "createDatetime", "updateTicket"
+					const mutationName = `${mutationType.toLowerCase()}Espresso${typeName}`;
+					// Example result: { data: { deletePrice: { price : {...} } } }
+					const path = ['data', mutationName, `espresso${typeName}`];
+					const entity = pathOr<any>({}, path, result);
 
-				onUpdate({ proxy, entity });
+					onUpdate({ proxy, entity });
+				};
 			};
-		};
 
-		const options = getMutationOptions();
-		return {
-			mutation: getMutation(),
-			...options,
-		};
-	}, []);
+			const options = getMutationOptions();
+			return {
+				mutation: getMutation(),
+				...options,
+			};
+		},
+		[getMutationHandler]
```

Comment:
`typeName` also needs to be included:

![image](https://user-images.githubusercontent.com/1477453/80708341-3095d300-8af4-11ea-84e8-756c52ace857.png)

Reply:
Updated